File: Datastruct/stack/stack.py
if __name__=="__main__":
    m = int(input())
    stk = []

    for i in range(m):
        op = input().split()
        if op[0] == "push":
            x = int(op[1])
            stk.append(x)
        elif op[0] == "pop":
            stk.pop()
        elif op[0] == "empty":
            if stk:
                print("NO")
            else:
                print("YES")
        elif op[0] == "query":
            if stk:
                print(stk[-1])

File: Datastruct/stack/stack.cpp
#include <iostream>

using namespace std;

//数组模拟栈

const int N = 1e5 + 10;

int stk[N], tt;

void push(int x) { stk[++tt] = x; }

void pop() { tt--; }

int top() { return stk[tt]; }

bool empty() { return tt == 0; }

int main() {
  int m;
  cin >> m;
  while (m--) {
    string op;
    int x;
    cin >> op;
    if (op == "push") {
      cin >> x;
      push(x);
    } else if (op == "pop") {
      pop();
    } else if (op == "empty") {
      cout << (empty() ? "YES" : "NO") << endl;
    } else {
      cout << top() << endl;
    }
  }

  return 0;
}
